### PR TITLE
improvement(node setup): Configure a hybrid RAID on GCE

### PR DESCRIPTION
**Pull request by yarongilor**
Use a custom script to replace Scylla perftune.py,
	That configures a hybrid RAID-1, using mdadm devices.

This PR is dependant on https://github.com/scylladb/scylla-qa-internal/pull/30
Task: https://github.com/scylladb/qa-tasks/issues/21
[Test plan](https://docs.google.com/document/d/1omGkONrwtm165_8fWf8r8-_gMl9753TW9uII8P4tjDI/edit?usp=sharing)

#### Testing
- [x] - https://jenkins.scylladb.com/job/scylla-staging/job/yarongilor/job/gemini-3h-with-nemesis-test-hybrid-raid-n2-highmem-32/7/
- [x] [argus job](https://argus.scylladb.com/test/574507a7-22f2-4df6-a854-3eb3496c832b/runs?additionalRuns%5B%5D=c09b5b84-59ca-463e-8d01-f0412b620fce)

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by yarongilor** [comment]
have to cherry pick the new param definitions as well (gce_setup_hybrid_raid)

**Review by fruch — DISMISSED** [review]
LGTM

File: sdcm/cluster.py [hunk 1]
```diff
@@ -4561,6 +4561,26 @@ def node_setup(self, node: BaseNode, verbose: bool = False, timeout: int = 3600)
...
+                    gce_pd_ssd_disk_size_db = self.params.get('gce_pd_ssd_disk_size_db')
+                    if not (gce_n_local_ssd_disk_db > 0 and gce_pd_ssd_disk_size_db > 0):
+                        msg = f"Hybrid RAID cannot be configured without NVMe ({gce_n_local_ssd_disk_db}) " \
+                              f"and PD-SSD ({gce_pd_ssd_disk_size_db})"
+                        raise ValueError(msg)
+
+                    # The script to configure a hybrid RAID is named "hybrid_raid.py" and located on the private repository.
+                    hybrid_raid_script = "hybrid_raid.py"
+                    target_path = os.path.join("/tmp", hybrid_raid_script)
+                    node.remoter.send_files(src=f"./scylla-qa-internal/custom_d1/{hybrid_raid_script}", dst=target_path)
```
Comment (fruch) [inline sdcm/cluster.py:4575]: I would suggest using absolute paths and not relative 

We've been bit by that before

**Comment by fruch** [comment]
@yarongilor I don't need more runs gemini right now, I doesn't tell us anything important right now

File: sdcm/cluster.py (hunk 1 above)
Comment (yarongilor) [reply sdcm/cluster.py:4575]: it was already tested and ran ok with: f"scylla-qa-internal/custom_d1/{hybrid_raid_script}"

i'll push that as well.

Comment (fruch) [reply sdcm/cluster.py:4575]: again i'm asking absolute paths

File: sdcm/cluster.py [hunk 2]
```diff
@@ -4561,6 +4561,26 @@ def node_setup(self, node: BaseNode, verbose: bool = False, timeout: int = 3600)
...
+                    gce_pd_ssd_disk_size_db = self.params.get('gce_pd_ssd_disk_size_db')
+                    if not (gce_n_local_ssd_disk_db > 0 and gce_pd_ssd_disk_size_db > 0):
+                        msg = f"Hybrid RAID cannot be configured without NVMe ({gce_n_local_ssd_disk_db}) " \
+                              f"and PD-SSD ({gce_pd_ssd_disk_size_db})"
+                        raise ValueError(msg)
+
+                    # The script to configure a hybrid RAID is named "hybrid_raid.py" and located on the private repository.
+                    hybrid_raid_script = "hybrid_raid.py"
+                    target_path = os.path.join("/tmp", hybrid_raid_script)
+                    node.remoter.send_files(src=f"scylla-qa-internal/custom_d1/{hybrid_raid_script}", dst=target_path)
```
Comment (fruch) [inline sdcm/cluster.py:4575]: something like that would be better
```suggestion
                    node.remoter.send_files(src=f"{get_sct_root_path()}/scylla-qa-internal/custom_d1/{hybrid_raid_script}", dst=target_path)
```

we can't really control in which directory we are, some code might change it without us knowing about it.

**Comment by yarongilor** [comment]
A [test](https://argus.scylladb.com/test/574507a7-22f2-4df6-a854-3eb3496c832b/runs?additionalRuns[]=c09b5b84-59ca-463e-8d01-f0412b620fce) is running ok with c-s read/write load and nemesis.

File: sdcm/cluster.py (hunk 2 above)
Comment (yarongilor) [reply sdcm/cluster.py]: @fruch , fixed.

**Comment by fruch** [comment]
> A [test](https://argus.scylladb.com/test/574507a7-22f2-4df6-a854-3eb3496c832b/runs?additionalRuns%5B%5D=c09b5b84-59ca-463e-8d01-f0412b620fce) is running ok with c-s read/write load and nemesis.

seems like there was a timeout, and the case was stopped.

```
2023-10-24 13:11:38.272: (CassandraStressLogEvent Severity.CRITICAL) period_type=one-time event_id=c925d1fd-efa5-46c4-a941-65d43bca1e50: type=OperationOnKey regex=Operation x10 on key\(s\) \[ line_number=939 node=Node longevity-3h-custom-d1-custom-p-loader-node-c09b5b84-0-1 [34.75.105.4 | 10.142.0.55] (seed: False)
java.io.IOException: Operation x10 on key(s) [50353637334f374e3330]: Error executing: (ReadTimeoutException): Cassandra timeout during read query at consistency QUORUM (2 responses were required but only 1 replica responded). In case this was generated during read repair, the consistency level is not representative of the actual consistency.
```

But i'll assume for now it's not optimized of the speed of the disk

**Comment by fruch** [comment]
> > A [test](https://argus.scylladb.com/test/574507a7-22f2-4df6-a854-3eb3496c832b/runs?additionalRuns%5B%5D=c09b5b84-59ca-463e-8d01-f0412b620fce) is running ok with c-s read/write load and nemesis.
> 
> seems like there was a timeout, and the case was stopped.
> 
> ```
> 2023-10-24 13:11:38.272: (CassandraStressLogEvent Severity.CRITICAL) period_type=one-time event_id=c925d1fd-efa5-46c4-a941-65d43bca1e50: type=OperationOnKey regex=Operation x10 on key\(s\) \[ line_number=939 node=Node longevity-3h-custom-d1-custom-p-loader-node-c09b5b84-0-1 [34.75.105.4 | 10.142.0.55] (seed: False)
> java.io.IOException: Operation x10 on key(s) [50353637334f374e3330]: Error executing: (ReadTimeoutException): Cassandra timeout during read query at consistency QUORUM (2 responses were required but only 1 replica responded). In case this was generated during read repair, the consistency level is not representative of the actual consistency.
> ```
> 
> But i'll assume for now it's not optimized of the speed of the disk


I'll wait for log collection, to see the monitor of that run

**Review by fruch — APPROVED** [review]
LGTM

**Comment by yarongilor** [comment]
@fruch , the test also failed similarly with c-s for read timeout (after adding a new node). should i retest with more nodes?  6 instead of 3?

**Comment by fruch** [comment]
> @fruch , the test also failed similarly with c-s for read timeout (after adding a new node). should i retest with more nodes? 6 instead of 3? or run 3 nodes without nemesis? The monitor shows a new added node is going crazy with 60 compactions: image The cluster is in a very unbalanced mode where the new-added node (10.142.0.74) has both much higher read and write load, right before the c-s timeout failure: image No CPU/memory issues was found.

the issue is probably disk
and we'll need to configure the io_properties.yaml correctly, iotune doesn't help us with network disks
the need to be tuned manually, and we are missing that part

**Comment by fruch** [comment]
> > > @fruch , the test also failed similarly with c-s for read timeout (after adding a new node). should i retest with more nodes? 6 instead of 3? or run 3 nodes without nemesis? The monitor shows a new added node is going crazy with 60 compactions: image The cluster is in a very unbalanced mode where the new-added node (10.142.0.74) has both much higher read and write load, right before the c-s timeout failure: image No CPU/memory issues was found.
> > 
> > 
> > the issue is probably disk and we'll need to configure the io_properties.yaml correctly, iotune doesn't help us with network disks the need to be tuned manually, and we are missing that part
> 
> @fruch, comparing io_properties shows -
> 
> ```
> disks:
>   - mountpoint: /var/lib/scylla
>     read_iops: 2300000
>     read_bandwidth: 9405726720
>     write_iops: 22500
>     write_bandwidth: 754974720
>     duplex: true
> ```
> 
> versus:
> 
> ```
> (read_iops=510000, write_iops=5400, read_bandwidth=2076180480, write_bandwidth=94371840))
> ```
> 
> But the number of disks is different

but their capabilities are the same regardless of how may we have

**Pull request merged**